### ISSUE TITLE
Create Auto Update Feature.

### ISSUE THREAD

**Issue by PaycoinMaster** [PaycoinFoundation/paycoin#298]
Build Auto Update into QT and Daemon So users don't have to have to download and install the wallets them self.

**Comment by RdeWilde** [comment]
\o/

**Comment by ligerzero459** [comment]
:+1: x1000

**Comment by mitchellcash** [comment]
I agree I would like this but am definitely concerned regarding Bitcoin Core Dev's comments around auto-update:

**laanwj commented on 24 Apr 2014**
```
Closing this: auto-update is never going to happen for security reasons. We already send out alerts when upgrading is absolutely necessary.
```

It seems that by implementing auto-update security holes are introduced, understandably, if you could perform a MITM attack and force a user to download a 'key stealing' version of the client we are going to have a lot of issues on our hand.

I'm not sure if this is the reason laanwj made his comment but I assume it's only one of many different reasons why this could turn bad. Even though I would love to see this, I think we best leave auto-update behind. Just quickly I can think of another few reasons why this could possibly go bad.

Although I am torn... as much as I want this I think I can convince myself to a few different reasons why it might not be safe.

Would love other comments.

**Comment by ligerzero459** [comment]
You know, I agree. There are a ton of security concerns that come with auto-updates. The risks outweigh the rewards in this case

**Comment by HyperCommNet** [comment]
It would have to be a feature they have to opt in to to use it. So it is not auto enabled. They would have to turn it on with a Popup telling them the risks and if they accept it.

**Comment by dbroyhill** [comment]
A notice of update is good enough?

**Comment by mitchellcash** [comment]
@ligerzero459 my thoughts exactly *the risks outweigh the rewards*.

@dbroyhill which we do have if it is urgently needed only at this stage.

So my thoughts are not to auto-update but we could attempt to display a message in the about box if you are out of date.

Potentially it pings paycoin.com for a version number check weekly and if the version doesn't match it displays a warning there (sort of like Firefox but without the actual auto-update):

![](https://dl.dropboxusercontent.com/u/37902134/GitHub/PaycoinFoundation%3Apaycoin/%23298_0.png)

**Comment by ligerzero459** [comment]
Perhaps also a pop-up message in the OS' message center to alert users that might not click into their wallets every day? Similar to how the OS alerts users of transactions or stakes

**Comment by IngCr3at1on** [comment]
I have the version querying and message displaying (using the coin alert message box) in a private branch. There were some issues with running a HTTP request to github initially. Obviously their API requires HTTPS which meant I couldn't use the default coin socket handling that's used for the GET requests in net.cpp so I wrote something using boost asio (very similar to the RPC handler). I haven't pushed it cause there's some memory errors in valgrind but from what I've been able to tell the errors are related directly to Boost Asio SSL. I might push my test code (not related to the coin) somewhere if anyone wants to play with it.

**Comment by mitchellcash** [comment]
I'm interested in taking a look

**Comment by IngCr3at1on** [comment]
@mitchellcash https://github.com/IngCr3at1on/boost-request

**Comment by IngCr3at1on** [comment]
@mitchellcash I linked you my test app (which is what I was running the valgrind tests on and noticed the Boost Asio SSL issues; I never asked if you actually wanted to see the coin code.
